lib/ansible/plugins/lookup/azure_keyvault.py

```diff
@@ -0,0 +1,140 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = """
+  lookup: azure_keyvault
+  author: James Johnson <james.johnson@hmcts.net>
+  short_description: Retrieve secrets from an Azure Key Vault.
+  requirements:
+    - azure-keyvault (python library)
+  description:
+    - Retrieve secrets from an Azure Key-Vault.
+  options:
+    secret_name:
+      description: The name of the Azure secret requested.
+      required: True
+    vault_uri:
+      description: The URI of the Azure Key Vault to query.
+      required: True
+      env:
+        - name: AZURE_VAULT_URI
+    secret_version:
+      description: The version of Azure secret.
+      default: 'latest'
+    client_id:
+      description: The Azure Client ID (also known as Application ID).
+      required: True
+      env:
+        - name: AZURE_CLIENT_ID
+    secret:
+      description: Your Azure Active Directory Service Principal AppId.
+      required: True
+      env:
+        - name: AZURE_SECRET
+    tenant:
+      description: Your Azure Active Directory Application Key.
+      required: True
+      env:
+        - name: AZURE_TENANT
+"""
+
+EXAMPLES = """
+- name: Return a secret, specifying minimal required parameters.
+  debug:
+    msg: "{{ lookup('azure_keyvault', 'secret_name=someSecretName') }}"
+
+- name: Return latest version of a secret from a specified vault_uri.
+  debug:
+    msg: "{{ lookup('azure_keyvault', 'secret_name=someSecretName vault_uri=https://yourvault.vault.azure.net/') }}"
+
+- name: Return a specific version of a secret with AZURE_VAULT_URI environment variable set.
+  debug:
+    msg: "{{ lookup('azure_keyvault', 'secret_name=someSecretName secret_version=169591fbe36742beb109478459f426ce') }}"
+
+ - name: Return a specific version of a secret from a specified vault_uri.
+  debug:
+    msg: "{{ lookup('azure_keyvault', 'secret_name=someSecretName secret_version=169591fb vault_uri=https://anothervault.vault.azure.net/') }}"
+"""
+
+RETURN = """
+_raw:
+  description:
+    - the secret requested.
+"""
+
+import os
+
+from ansible.errors import AnsibleError
+from ansible.plugins.lookup import LookupBase
+
+try:
+    from azure.common.credentials import ServicePrincipalCredentials
+    from azure.keyvault import KeyVaultClient
+
+except ImportError:
+    raise AnsibleError(
+        "Please run 'pip install azure-keyvault' to use the azure_keyvault lookup module.")
+
+
+class LookupModule(LookupBase):
+
+    def run(self, terms, variables=None, **kwargs):
+
+        ret = []
+
+        accepted_params = {
+            'secret_name': None,
+            'vault_uri': os.environ.get('AZURE_VAULT_URI', None),
```

Comment:
This should use the existing `AzureRMAuth ` rather than requiring these keyvault vars only. Then auth using alternative ansible support methods works too. https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/azure_rm_common.py#L971

This is how the other azure plugins work already (e.g. inventory plugin https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/inventory/azure_rm.py#L274 )